tree/tree/Extension/Extension+UITableViewDelegate.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  Extension+UITableViewDelegate.swift
+//  tree
+//
+//  Created by Hyeontae on 14/02/2019.
+//  Copyright © 2019 gardener. All rights reserved.
+//
+
+import UIKit
+import NetworkFetcher
+
+enum CustomUIContextualActionState {
+    case scrap
+    case markAsRead
+    case delete
+}
+
+extension UITableViewDelegate {
+    func customUIContextualAction(
+        _ type: CustomUIContextualActionState,
+        _ articleData: Article?,
+        _ imageData: Data?,
+        _ articleIdentifier: String?,
+        _ articleType: ScrappedArticleType?,
+        completion: @escaping (Bool) -> Void
+        ) -> UIContextualAction {
+        switch type {
+        case .scrap:
+            guard let articleData = articleData else { return UIContextualAction() }
+            var isScrapped: Bool = true
+            var scrppaedArticle: ScrappedArticle?
+            ScrapManager.articleIsScrapped(uri: articleData.uri) { (articleScrapped, article) in
+                isScrapped = articleScrapped
+                scrppaedArticle = article
+            }
+            let scrapAction = UIContextualAction(
+                style: .normal,
+                title: nil) { (_, _, success) in
+                    if !isScrapped {
+                        ScrapManager.scrapArticle(
+                            .search ,
+                            articleStruct:
+                            SearchedArticleStruct(articleData: articleData, imageData: imageData)
+                        )
+                    } else {
+                        if let tempscrppaedArticle = scrppaedArticle {
+                            ScrapManager.removeArticle(tempscrppaedArticle)
+                        }
+                    }
+                    guard
+                        let appDelegate =
+                            UIApplication.shared.delegate as? AppDelegate,
+                        let scrapViewController =
+                            appDelegate.scrapViewController as? ScrapViewController
+                        else { return }
+                    scrapViewController.scrappedArticles = ScrapManager.fetchArticles()
+                    scrapViewController.setupScrapBadgeValue()
+                    completion(true)
+                    success(true)
+            }
+            if isScrapped {
+                scrapAction.image = UIImage(named: "upload.png")
```

Comment:
[Asset 적용]
앱에 이미지를 넣을때는 `Assets.xcassets`에 이미지를 넣고 불러올 때는 확장자를 제외한 Asset 이름만으로 불러오는 것이 좋습니다.
`.xcassets` 파일은 기본적으로 제공되는 `Assets.xcassets` 외에도 필요에 의해 추가로 생성 가능합니다.

```suggestion
                (xcassets 파일에 이미지를 넣은 후)
                scrapAction.image = UIImage(named: "upload")
```